### PR TITLE
use ocs discover endpoint to test remote

### PR DESCRIPTION
**Pull request by icewind1991**
check `owncloud/ocs-provider` instead of `owncloud/status.php`

status.php is still checked as a fallback for versions <8.1

cc @PVince81 @LukasReschke

### CODE REVIEW

File: apps/files_sharing/lib/external/storage.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
  *
  */
 
-namespace OCA\Files_Sharing\External;
+namespace test;
```
Comment (PVince81) [inline apps/files_sharing/lib/external/storage.php]: huh?

**Comment by PVince81** [comment]
@icewind1991 did you test it ?

```
{"reqId":"nguXOSoRx3HaRkqt8tE\/","remoteAddr":"127.0.0.1","app":"core","message":"storage backend \\OCA\\Files_Sharing\\External\\Storage not found","level":3,"time":"2016-02-29T14:59:47+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/"}
{"reqId":"UnJjSOGdscDHOBIjnnX9","remoteAddr":"127.0.0.1","app":"core","message":"storage backend \\OCA\\Files_Sharing\\External\\Storage not found","level":3,"time":"2016-02-29T14:59:48+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/admin\/26"}
{"reqId":"pqqG7aZRH9qE0zOYDn65","remoteAddr":"127.0.0.1","app":"core","message":"storage backend \\OCA\\Files_Sharing\\External\\Storage not found","level":3,"time":"2016-02-29T14:59:48+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/ajax\/getstoragestats.php?dir=%2F"}
```

Steps:

1. Setup OC A with 9.0.0 beta2
1. Setup OC B with this branch
1. From OC A, share a folder with "admin@OC_B"
1. In OC B, accept the share

If OC B is on master: works fine.
If OC B is on this branch: fails with the errors above

File: apps/files_sharing/lib/external/storage.php (hunk 1 above)
Comment (icewind1991) [inline apps/files_sharing/lib/external/storage.php]: no idea how that happened

**Comment by icewind1991** [comment]
please retest

**Comment by PVince81** [comment]
Tested, connecting to a remote share on OC 9.0 beta2 and OC 8.0. Both worked.
Also tested "storage not available case" when target server is in maintenance mode.

:+1:

**Comment by PVince81** [comment]
Something fishy here:
```
10:15:46 1) OCA\Files_Sharing\Controllers\ExternalShareControllerTest::testRemoteWithValidHttps
10:15:46 Failed asserting that two objects are equal.
10:15:46 --- Expected
10:15:46 +++ Actual
10:15:46 @@ @@
10:15:46  OCP\AppFramework\Http\DataResponse Object (
10:15:46 -    'data' => 'https'
10:15:46 +    'data' => false
10:15:46      'headers' => Array (...)
10:15:46      'cookies' => Array ()
10:15:46      'status' => 200
10:15:46      'lastModified' => null
10:15:46      'ETag' => null
10:15:46      'contentSecurityPolicy' => OCP\AppFramework\Http\ContentSecurityPolicy Object (...)
10:15:46  )
10:15:46 
10:15:46 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/files_sharing/tests/controller/externalsharecontroller.php:116
10:15:46 
10:15:46 2) OCA\Files_Sharing\Controllers\ExternalShareControllerTest::testRemoteWithWorkingHttp
10:15:46 Failed asserting that two objects are equal.
10:15:46 --- Expected
10:15:46 +++ Actual
10:15:46 @@ @@
10:15:46  OCP\AppFramework\Http\DataResponse Object (
10:15:46 -    'data' => 'http'
10:15:46 +    'data' => 'https'
10:15:46      'headers' => Array (...)
10:15:46      'cookies' => Array ()
10:15:46      'status' => 200
10:15:46      'lastModified' => null
10:15:46      'ETag' => null
10:15:46      'contentSecurityPolicy' => OCP\AppFramework\Http\ContentSecurityPolicy Object (...)
10:15:46  )
10:15:46 
10:15:46 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/files_sharing/tests/controller/externalsharecontroller.php:136
10:15:46 
10:15:46 3) OCA\Files_Sharing\Controllers\ExternalShareControllerTest::testRemoteWithInvalidRemote
10:15:46 Expectation failed for method name is equal to <string:newClient> when invoked 2 time(s).
10:15:46 Method was expected to be called 2 times, actually called 6 times.
```

File: apps/files_sharing/lib/external/storage.php [hunk 2]
```diff
@@ -218,20 +218,26 @@ public function file_exists($path) {
...
 	 *
 	 * @return bool
 	 */
 	protected function testRemote() {
 		try {
-			$result = file_get_contents($this->remote . '/status.php');
-			$data = json_decode($result);
-			return is_object($data) and !empty($data->version);
+			return $this->testRemoteUrl($this->remote . '/ocs-provider/index.php')
+				|| $this->testRemoteUrl($this->remote . '/ocs-provider')
```
Comment (LukasReschke) [inline apps/files_sharing/lib/external/storage.php]: Requires the trailing slash

File: apps/files_sharing/lib/controllers/externalsharescontroller.php [hunk 3]
```diff
@@ -124,9 +129,17 @@ protected function testUrl($remote) {
 	 * @return DataResponse
 	 */
 	public function testRemote($remote) {
-		if ($this->testUrl('https://' . $remote . '/status.php')) {
+		if (
+			$this->testUrl('https://' . $remote . '/ocs-provider') ||
```
Comment (LukasReschke) [inline apps/files_sharing/lib/controllers/externalsharescontroller.php]: Requires the trailing slash

File: apps/files_sharing/lib/controllers/externalsharescontroller.php [hunk 4]
```diff
@@ -124,9 +129,17 @@ protected function testUrl($remote) {
...
+			$this->testUrl('https://' . $remote . '/ocs-provider/index.php') ||
+			$this->testUrl('https://' . $remote . '/status.php', true)
+		) {
 			return new DataResponse('https');
-		} elseif ($this->testUrl('http://' . $remote . '/status.php')) {
+		} elseif (
+			$this->testUrl('http://' . $remote . '/ocs-provider') ||
```
Comment (LukasReschke) [inline apps/files_sharing/lib/controllers/externalsharescontroller.php]: Requires the trailing slash

**Comment by LukasReschke** [comment]
I fixed the tests, :+1: for the code changes from my side.

**Comment by karlitschek** [comment]
:+1: makes sense. Must have for 9.0

**Comment by karlitschek** [comment]
changes it to sev1 so that we don't forget to merge this.

**Comment by LukasReschke** [comment]
For the record: I did some more changes to this and in combination with https://github.com/owncloud/core/pull/22681 I'm able to receive shares from @Pydio.

Once https://github.com/pydio/pydio-core/issues/1066 is fixed it should also work the other way around.

**Comment by PVince81** [comment]
:+1: for @LukasReschke's changes

**Pull request merged**